src/ui/pages/CreateSSIAgent/CreateSSIAgent.tsx

```diff
@@ -0,0 +1,226 @@
+import { updateReduxState } from "../../../store/utils";
+import { CustomInput } from "../../components/CustomInput";
+import { ErrorMessage } from "../../components/ErrorMessage";
+import { PageFooter } from "../../components/PageFooter";
+import { PageHeader } from "../../components/PageHeader";
+import { OperationType } from "../../globals/types";
+import { useAppIonRouter } from "../../hooks";
+import "./CreateSSIAgent.scss";
+import {
+  clearSSIAgent,
+  getSSIAgent,
+  setBootUrl,
+  setConnectUrl,
+} from "../../../store/reducers/ssiAgent";
+import { isValidHttpUrl } from "../../utils/urlChecker";
+import { TermsModal } from "../../components/TermsModal";
+import { Agent } from "../../../core/agent/agent";
+import { ScrollablePageLayout } from "../../components/layout/ScrollablePageLayout";
+
+const CreateSSIAgent = () => {
+  const pageId = "create-ssi-agent";
+  const ssiAgent = useAppSelector(getSSIAgent);
+
+  const stateCache = useAppSelector(getStateCache);
+  const ionRouter = useAppIonRouter();
+  const dispatch = useAppDispatch();
+  const [connectUrlInputTouched, setConnectUrlTouched] = useState(false);
+  const [bootUrlInputTouched, setBootUrlInputTouched] = useState(false);
+  const [openInfo, setOpenInfo] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [hasMismatchError, setHasMismatchError] = useState(false);
+
+  const setTouchedConnectUrlInput = () => {
+    setConnectUrlTouched(true);
+  };
+
+  const setTouchedBootUrlInput = () => {
+    setBootUrlInputTouched(true);
+  };
+
+  const validBootUrl = useMemo(() => {
+    return ssiAgent.bootUrl && isValidHttpUrl(ssiAgent.bootUrl);
+  }, [ssiAgent]);
+
+  const validConnectUrl = useMemo(() => {
+    return ssiAgent.connectUrl && isValidHttpUrl(ssiAgent.connectUrl);
+  }, [ssiAgent]);
+
+  const displayBootUrlError =
+    bootUrlInputTouched &&
+    ssiAgent.bootUrl &&
+    !isValidHttpUrl(ssiAgent.bootUrl);
+
+  const displayConnectUrlError =
+    connectUrlInputTouched &&
+    ssiAgent.connectUrl &&
+    !isValidHttpUrl(ssiAgent.connectUrl);
+
+  const validated = validBootUrl && validConnectUrl && !hasMismatchError;
+
+  const handleClearState = () => {
+    dispatch(clearSSIAgent());
+  };
+
+  const handleValidate = async () => {
+    setLoading(true);
+    try {
+      await Agent.agent.bootAndConnect({
+        bootUrl: ssiAgent.bootUrl || "",
+        url: ssiAgent.connectUrl || "",
+      });
+
+      const { nextPath, updateRedux } = getNextRoute(RoutePath.SSI_AGENT, {
+        store: { stateCache },
+      });
+
+      updateReduxState(
+        nextPath.pathname,
+        {
+          store: { stateCache },
+        },
+        dispatch,
+        updateRedux
+      );
+
+      ionRouter.push(nextPath.pathname, "forward", "push");
+      handleClearState();
+    } catch (e) {
+      const errorMessage = (e as Error).message;
+      if (Agent.KERIA_BOOTED_ALREADY_BUT_CANNOT_CONNECT === errorMessage) {
```

Comment:
`Agent.KERIA_BOOT_FAILED` can also happen if the boot URL is a valid URL but is not a KERIA agent (or just rejects us). We need to handle this too - can you check with Robert what error to show in this case?